generate-self-signed-certificates.md

```diff
@@ -1,156 +1,176 @@
-                    "key encipherment",
-                    "server auth",
-                    "client auth"
-                ]
-            },
-            "client": {
-                "expiry": "43800h",
-                "usages": [
-                    "signing",
-                    "key encipherment",
-                    "client auth"
-                ]
-            }
-        }
-    }
-}
-```
-
-Edit `ca-csr.json` according to your need:
-
-```json
-{
-    "CN": "My own CA",
-    "key": {
-        "algo": "rsa",
-        "size": 2048
-    },
-    "names": [
-        {
-            "C": "CN",
-            "L": "Beijing",
-            "O": "PingCAP",
-            "ST": "Beijing"
-        }
-    ]
-}
-```
-
-### Generate the CA certificate
-
-```bash
-cfssl gencert -initca ca-csr.json | cfssljson -bare ca -
-```
-
-The command above generates the following files:
-
-```bash
-ca-key.pem
-ca.csr
-ca.pem
-```
-
-### Generate the server certificate
-
-The IP address of all components and `127.0.0.1` are included in `hostname`.
-
-```bash
-echo '{"CN":"tidb-server","hosts":[""],"key":{"algo":"rsa","size":2048}}' | cfssl gencert -ca=ca.pem -ca-key=ca-key.pem -config=ca-config.json -profile=server -hostname="172.16.10.1,172.16.10.2,127.0.0.1" - | cfssljson -bare tidb-server
-
-echo '{"CN":"tikv-server","hosts":[""],"key":{"algo":"rsa","size":2048}}' | cfssl gencert -ca=ca.pem -ca-key=ca-key.pem -config=ca-config.json -profile=server -hostname="172.16.10.4,172.16.10.5,172.16.10.6,127.0.0.1" - | cfssljson -bare tikv-server
-
-echo '{"CN":"pd-server","hosts":[""],"key":{"algo":"rsa","size":2048}}' | cfssl gencert -ca=ca.pem -ca-key=ca-key.pem -config=ca-config.json -profile=server -hostname="172.16.10.1,172.16.10.2,172.16.10.3,127.0.0.1" - | cfssljson -bare pd-server
-```
-
-The command above generates the following files:
-
-```Bash
-tidb-server-key.pem     tikv-server-key.pem      pd-server-key.pem
-tidb-server.csr         tikv-server.csr          pd-server.csr
-tidb-server.pem         tikv-server.pem          pd-server.pem
-```
-
-### Generate the client certificate
-
-```bash
-echo '{"CN":"client","hosts":[""],"key":{"algo":"rsa","size":2048}}' | cfssl gencert -ca=ca.pem -ca-key=ca-key.pem -config=ca-config.json -profile=client -hostname="" - | cfssljson -bare client
-```
-
-The command above generates the following files:
-
-```bash
-client-key.pem
-client.csr
-client.pem
-```
+- client certificate: used to authenticate the clients from PD, TiKV and TiDB, such as `pd-ctl`, `tikv-ctl`
+
+### Issue certificates to TiKV instances
+
+To issue a certificate to a TiKV instance, the steps are as follows:
+
+1. Generate the private key corresponding to the certificate:
+
+    {{< copyable "shell-regular" >}}
+
+    ```bash
+    openssl genrsa -out tikv.key 2048
+    ```
+
+2. Make a copy of the OpenSSL configuration template file (Refer to the actual location of your template file because it may have more than one location):
```

Comment:
```suggestion
2. Make a copy of the OpenSSL configuration template file (Refer to the actual location of your template file because it might have more than one location):
```